src/oop/programs_with_multiple_classes/medium/q3/MainTest.java

```diff
@@ -0,0 +1,155 @@
+package oop.programs_with_multiple_classes.medium.q3;
+
+import global.BaseTest;
+import global.ObjectTest;
+import global.tools.TestOption;
+import global.variables.Clause;
+import global.variables.clauses.NewLine;
+import global.variables.clauses.StringLiteral;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import java.util.stream.Stream;
+
+import static global.tools.CustomAssertions._assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class MainTest extends BaseTest {
+    // Parsons
+
+    public ObjectTest pencil;
+    public ObjectTest eraser;
+
+    @BeforeEach
+    public void setup() {
+        String pencilClassString = "oop.programs_with_multiple_classes.medium.q3.Pencil";
+        String eraserClassString = "oop.programs_with_multiple_classes.medium.q3.Eraser";
+        pencil = new ObjectTest(pencilClassString);
+        eraser = new ObjectTest(eraserClassString);
+    }
+
+    public Clause[] testSentence() {
+        TestOption.incorrectStructureErrorMessage =
+                "Your program does not correctly print out the toString for each initialized object in the TestRoom class.";
+        return new Clause[]{
+                new StringLiteral("Pencil 1: Pencil\\{hardness = 2, isForSchool = false}"),
+                new NewLine(),
+                new StringLiteral("Pencil 2: Pencil\\{hardness = 1, isForSchool = true}"),
+                new NewLine(),
+                new StringLiteral("Eraser 1: Eraser\\{size = 10.5}"),
+                new NewLine(),
+                new StringLiteral("Eraser 2: Eraser\\{size = 4.3}"),
+                new NewLine()
+        };
+    }
+
+    public void runMain() {
+        TestPencilCase.main(new String[0]);
+    }
+
+    @Test
+    public void pencilClassHasCorrectAttributes() {
+        String incorrectFieldMessage = "Your Pencil class is missing a required field.";
+        String incorrectModifierMessage = "One of your Pencil class attributes does not have the correct modifier.";
```

Comment:
To make it easier to vary on the website, you should remove the class, field, and method names from your Strings and convert them to a final variable at the top of the class. That way, you can do something like 
"One of your " + className + " class attributes does not have the correct modifier."

Reply:
Then the person who enters the question on the website does not have to change every single instance of the name, just the one at the top of the class